client/src/PlayerCommands.cpp

```diff
@@ -44,31 +44,31 @@ bool PlayerCommands::isMovingRight() const {
 PlayerCommands::PlayerCommands() : movingUp(false), movingDown(false), movingLeft(false), movingRight(false),
                                    quadTreeDebug(false) {}
 
-Player *PlayerCommands::getPlayer() const {
+const std::weak_ptr<Player> &PlayerCommands::getPlayer() const {
     return player;
 }
 
-void PlayerCommands::setPlayer(Player *player) {
+void PlayerCommands::setPlayer(const std::weak_ptr<Player> &player) {
     PlayerCommands::player = player;
 }
 
 void PlayerCommands::updatePlayer(sf::Time deltaTime) {
     sf::Vector2f movement(0.f, 0.f);
     if (movingUp)
-        movement.y -= player->getSpeed();
+        movement.y -= player.lock()->getSpeed();
     if (movingDown)
-        movement.y += player->getSpeed();
+        movement.y += player.lock()->getSpeed();
```

Comment:
Ok, I will fix that.
Therefore, each time there are several call to the smart pointer in the same block I should use a temporary shared_ptr.